From ed42ea59f8de06c1c15d1549585b794129a037b6 Mon Sep 17 00:00:00 2001
From: AD-Aleksandrov <aleksandr.aleksandrov@emlid.com>
Date: Thu, 30 Nov 2017 14:49:58 +0300
Subject: [PATCH] rtc: rtc-sun6i: Enable LOSC output gating

It's needed for peripheral devices, bluetooth device, for example:
Ampak AP6212(A): bluetooth doesn't work correctly without External Low Power Clock
---
 drivers/rtc/rtc-sun6i.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/rtc/rtc-sun6i.c b/drivers/rtc/rtc-sun6i.c
index 3d2216c..2c529a0 100644
--- a/drivers/rtc/rtc-sun6i.c
+++ b/drivers/rtc/rtc-sun6i.c
@@ -216,6 +216,10 @@ static void __init sun6i_rtc_clk_init(struct device_node *node)
 	writel(SUN6I_LOSC_CTRL_KEY | SUN6I_LOSC_CTRL_EXT_OSC,
 	       rtc->base + SUN6I_LOSC_CTRL);
 
+	/* Enable LOSC output gating, it's needed for peripheral devices */
+	writel(SUN6I_LOSC_OUT_GATING_EN,
+	       rtc->base + SUN6I_LOSC_OUT_GATING);
+
 	/* Yes, I know, this is ugly. */
 	sun6i_rtc = rtc;
 
-- 
2.7.4

